core/modules/xrdsvc/SsiProvider.cc

```diff
@@ -0,0 +1,148 @@
+/// implementation. Link this file when building a plugin to be used as
+/// ssi.svclib or oss.statlib.
+
+// Class header
+#include "xrdsvc/SsiProvider.h"
+
+// System headers
+#include <memory>
+#include <sstream>
+#include <sys/types.h>
+
+// Third party headers
+#include "XrdSsi/XrdSsiCluster.hh"
+#include "XrdSsi/XrdSsiLogger.hh"
+
+// LSST headers
+#include "lsst/log/Log.h"
+
+// Qserv headers
+#include "global/ResourceUnit.h"
+#include "wpublish/ChunkInventory.h"
+#include "xrdsvc/XrdName.h"
+
+/******************************************************************************/
+/*                               G l o b a l s                                */
+/******************************************************************************/
+
+XrdSsiProvider *XrdSsiProviderServer = 
+                new  lsst::qserv::xrdsvc::SsiProviderServer;
+
+XrdSsiProvider *XrdSsiProviderLookup = XrdSsiProviderServer;
+  
+/******************************************************************************/
+/*                                  I n i t                                   */
+/******************************************************************************/
+  
+namespace lsst {
+namespace qserv {
+namespace xrdsvc {
+
+bool SsiProviderServer::Init(XrdSsiLogger* logP,  XrdSsiCluster* clsP, char const* cfgFn,
+          char const*   parms, int argc, char **argv) {
+
+    lsst::qserv::xrdsvc::XrdName x;
+
+// Establish our instance name (many different qservs may be running here).
+//
+    _name = x.getName();
+
+// Save the ssi logger as it places messages in anoher file than our log.
+//
+    _logSsi = logP;
+
+// Save the cluster object as we will need to use it to inform the cluster when
+// chunks come and go. We also can use it to schedule ourselves. The object or
+// its absence will indicate whether or not we need to provide any service other
+// than QueryResource().
+//
+    _cmsSsi = clsP;
+
+// We would process the configuration file (if present), any present parameters,
+// and the command line arguments. However, at the moment, we have nothing of
+// interest in any of these arguments. So, we ignore them.
+//
+
+// Herald our initialization
+//
+    LOG_INFO("SsiProvider initializing...");
+    _logSsi->Msg("Qserv", "Provider Initializing");
+
+// Initialize the inventory. We need to be able to handle QueryResource() calls
+// either in the data provider and the metadata provider (we can be either one).
+//
+    _chunkInventory.reset(new wpublish::ChunkInventory(_name));
+
+// If we are a data provider (i.e. xrootd) then we need to get the service
+// object. It will print the exported paths. Otherwise, we need to print them
+// here. This is a bit kludgy and should be corrected when we transition to a
+// single shared memory inventory object which should do this by itself.
+//
+    if(clsP && clsP->DataContext()) {
+        _service = std::unique_ptr<SsiService>(new SsiService(logP));
```

Comment:
few lines above there is `_chunkInventory.reset(...)` method used to change the value of unique_ptr, and here there is an assignment operator from temporary which does the same thing . For consistency I'd use one or another in the same method, `reset()` is shorter and more readable, so I like this better:

     _service.reset(new SsiService(logP));